### PR TITLE
Support dark mode

### PR DESCRIPTION
**Pull request by stevenguh**
I was not sure if this should go to `master` branch or `0.9.2-beta`

### CODE REVIEW

**Comment by noah-nuebling** [comment]
Hey there Steve, thank you for contributing! But have you tested if your solution works? As far as I know dark mode for Preference Panes is broken in Catalina unless System Integrity Protection is disabled. You can find reports of this all over if you search on Google. [This](https://www.noodlesoft.com/blog/2019/08/28/preference-panes-and-catalina/) is one example.

**Comment by stevenguh** [comment]
I think dark mode and Catalina's security problems are separate. I fond this change from another [repo](https://github.com/specialunderwear/Hosts.prefpane/pull/37/files) that implemented dark mode in prefpane. I tested this change with mac-mouse-fixer prefpane and a brand new prefpane with Catalina. It is true that from a brand new prefpane project from Xcode would not support dark mode right out the box unless the `NSRequiresAquaSystemAppearance` is set to `false`. See the last section of this document from Apple -- [Choosing a Specific Appearance for Your macOS App](https://developer.apple.com/documentation/appkit/nsappearancecustomization/choosing_a_specific_appearance_for_your_macos_app)

I am not a Mac developer by any means so I could be wrong. More testing is probably good to make sure this works as intended.

**Comment by stevenguh** [comment]
btw, I just want to say thank you for creating such an amazing tool. I have been using `0.9.2-beta` branch :)

**Comment by stevenguh** [comment]
I saw you adding it in `0.9.2-beta` branch, and I am going to close this.

**Comment by noah-nuebling** [comment]
Hey man thanks again for your contribution and I'm really happy you're liking the app! I was just playing around with the change you suggested and intended to merge your pull request at a later date. If you want to, I can undo the change and merge your pull request instead, so you get full 'pull-request-credit' if that's a thing. But of course I also plan on updating the Changelog with the new feature and releasing a new Beta version at some point and I'll credit you there as well. Oh and I could give you full access to the repo so you can push directly to all branches in the future. Would you be interested in that?

**Comment by stevenguh** [comment]
That's okay. I appreciate the effort to attribute. It's such a small feature, and I am just glad that it's on your radar. Thank you for offering the full access, I don't think if there is anything I can add at this point. I just want to help to make this better since I also use it.

By the way, there are few things that I realized but not sure if they are on your radar:
- Key customization on the UI for both global and in-app to scroll horizontally and zoom
- Dash mode like on MOS
- Zoom direction seems to be inverted. This can be experienced when use Maps.app. The zoom direction is inconsistent when scrolling with command key and without. There is a negative sign in `TouchSimulator.m` on the `magnification`, I assume that intentional. If that's the case, maybe that should be a config?

I might able to help some of these when I have time.

**Comment by noah-nuebling** [comment]
Hey there, I just published 0.9.2 Beta 4 and updated the changelog!

Regarding you suggestions:
> Key customization on the UI for both global and in-app to scroll horizontally and zoom

I like to keep the app as simple as possible wherever it makes sense. So in general, I only want to add an option if that option solves a problem. Can you think of such a problem?

> Dash mode like on MOS

This actually wasn't on my radar. Thanks for the heads up! Dash mode is a very cool feature, I really like it. But, I think Mac Mouse Fix already has a great solution for scrolling very long pages in fastScroll (idk what to call it). When you scroll your scroll wheel several times in the same direction in quick succession, scrolling will become much faster after a short time This was broken in the last few releases of 0.9.2 but I fixed it in Beta 4. I like the fastScroll feature because it requires no setup and you don't have to think about using it, it just happens naturally. It should also be much faster on extremely long pages compared to dash scroll (I just tested it and it can scroll the ["worlds longest website"](http://www.worldslongestwebsite.com/) in 28 touches of the scrollwheel haha). Getting to the top of a "normal" long page in a single scroll swipe is really awesome though. I'll definitely think about it some more.

> Zoom direction seems to be inverted.

My thinking behind this is the following:

To me the natural vertical direction is down - you start reading a page at the top and then you go to the bottom

The natural "distance direction" is from far to near - you usually go towards something, when you're looking at it. Zooming, to me, feels analogous to going towards something.

I feel like it makes sense and is slightly more intuitive for all the "natural directions" to align with one scrollwheel direction.

Another point to make is, that iOS Maps actually works this way. In iOS, you swipe __up__ to scroll down, and in iOS Maps you double-tap-and-drag __up__ to zoom in. 

So basically I think that macOS Maps just has it wrong. Also, again, I want to keep the app as simple as possible, and in this case, I don't think an option is necessary. I'm open to have my mind changed, though!

**Comment by stevenguh** [comment]
> I like to keep the app as simple as possible wherever it makes sense. So in general, I only want to add options if the lack of that option leads to a problem in some use-case. Can you think of such a situation?

I don't really have a great situation I can think of right now. I just thinks it's great to offer options since some people might want to use a different key for different functions.

> I like the fastScroll feature because it requires no setup and you don't have to think about using it, it just happens naturally. It should also be much faster on extremely long pages compared to dash scroll (I just tested it and it can scroll the "worlds longest website" in 28 touches of the scrollwheel haha)

I would love to experience that. I don't think it works on my end with latest `0.9.2 beta`. If it works as well as your describe, I think dash scroll might not offer much value.


> I feel like it makes sense and is slightly more intuitive for all the "natural directions" to align with one scrollwheel direction.

That reminds me of Apple's move to change the default scroll direction on trackpad. It might feel better for some, but they offer an option to switch back. I then started testing the apps I have with natural scroll direction unchecked and without mouse fix installed.

- Affinity Photo: option + scroll up = Zoom in
- Google Maps/Apple Maps: scroll up = Zoom in
- Firefox: Ctl + scroll up = Zoom in
- Fusion 360: scroll up = Zoom out (They offer an option to reserve zoom direction)

> Another point to make is, that iOS Maps actually works this way. In iOS, you swipe up to scroll down, and in iOS Maps you double-tap-and-drag up to zoom in.

I think that's a little bit unfair to compare to different input methods and user interfaces. However, I think it depends on how you think about the scroll direction. When you double tap and drag up, the iOS Map will zoom in; I can think of that as analog to a mouse wheel scroll up. Therefore, the current zoom is inverted.

Base the existing behavior, I believe the current zoom behavior is not the norm and can be unexpected for a mouse user. I am not saying there is a right "direction" because I don't think there could ever be one. Therefore, at best we should at least provide an option to the user to switch back even if you think that's "correct" direction.

One tidbit I learned this week is [Nielsen's 10 Heuristics](https://www.nngroup.com/articles/ten-usability-heuristics/), one of them being consistency, and I think that's applicable here.

**Comment by noah-nuebling** [comment]
You're making some great points. Now that you mention it, the way one-finger zooming works in iOS Maps also inconsistent with how it works in most other iOS apps, and I remember being irritated by it when I first used it. I totally agree that consistency is super important.

I will remove that minus sign at some point and I might provide an option to toggle zoom direction at some point in the future. Thanks! :)

**Comment by noah-nuebling** [comment]
> I would love to experience that.

It works for me on Beta 4. It starts making the scroll faster on the 4th scrollSwipe, but it only starts to become very noticeable on the 5th scrollSwipe, so you might wanna set the scrolling speed lower or visit the worlds longest website to test it. 

I'll look into having it trigger more easily, so it can help the user in more situations.

**Comment by noah-nuebling** [comment]
I changed it, so it triggers on the 3rd consecutive scroll swipe in the latest commit.

**Comment by stevenguh** [comment]
Thank you making the changes. As far as the fast scroll, I can barely trigger with my mouse. I can only consistent trigger fast scroll if I change `scrollSwipeThreshold_inTicks` to `2`. I assume that's probably variations in mouse manufacturer. Maybe this is another setting thing?
By the way, the fast scrolling is pretty cool when it works :)

**Comment by noah-nuebling** [comment]
Oh that's unfortunate! I didn't expect that to be an issue. What kind of mouse do you use?

And don't you experience any issues with triggering fast scroll accidentally when setting `scrollSwipeThreshold_inTicks ` so low?

**Comment by stevenguh** [comment]
I use an off brand vertical mouse. Maybe there can be some accential fast scrolling, but the exponential is small enough that I am fine. I personally like to scroll fast anyway.

**Comment by noah-nuebling** [comment]
Okay awesome, I'm glad that it works for you. I'll definitely check out your configuration, and I will put it into the next release if I like it.

**Comment by noah-nuebling** [comment]
By the way, you've been super helpful man, thank you for taking the time! :)
